File: src/main/java/forpleuvoir/hiirosakura/client/feature/event/eventinterface/impl/ClientPlayerInterface.java
package forpleuvoir.hiirosakura.client.feature.event.eventinterface.impl;

import forpleuvoir.hiirosakura.client.feature.event.eventinterface.IClientPlayerInterface;
import net.minecraft.client.network.ClientPlayerEntity;
import net.minecraft.util.registry.Registry;

/**
 * 客户端玩家接口实现
 *
 * @author forpleuvoir
 * <p>#project_name hiirosakura
 * <p>#package forpleuvoir.hiirosakura.client.feature.event.eventinterface.impl
 * <p>#class_name ClientPlayerInterface
 * <p>#create_time 2021/8/1 14:47
 */
public class ClientPlayerInterface implements IClientPlayerInterface {

    public ClientPlayerInterface(ClientPlayerEntity clientPlayerEntity) {
        this.player = clientPlayerEntity;
    }

    private final ClientPlayerEntity player;

    @Override
    public String getUUID() {
        return player.getUuidAsString();
    }

    @Override
    public String getName() {
        return player.getEntityName();
    }

    @Override
    public float getHealth() {
        return player.getHealth();
    }

    @Override
    public float getMaxHealth() {
        return player.getMaxHealth();
    }

    @Override
    public double getHealthProgress() {
        return getHealth() / getMaxHealth();
    }

    @Override
    public String getMainHandItemRegisterId() {
        return Registry.ITEM.getId(player.getMainHandStack().getItem()).toString();
    }

    @Override
    public String getMainHandItemTranslationKey() {
        return player.getMainHandStack().getItem().getTranslationKey();
    }

    @Override
    public String getMainHandItemDisplayName() {
        return player.getMainHandStack().getName().getString();
    }

    @Override
    public int getMainHandItemDurable() {
        return getMainHandItemMaxDurable() - player.getMainHandStack().getDamage();
    }

    @Override
    public int getMainHandItemMaxDurable() {
        return player.getMainHandStack().getMaxDamage();
    }

    @Override
    public double getMainHandItemDurableProgress() {
        return ((double) getMainHandItemDurable()) / player.getMainHandStack().getMaxDamage();
    }

    @Override
    public boolean isFireImmune() {
        return player.isFireImmune();
    }

    @Override
    public boolean isOnFire() {
        return player.isOnFire();
    }

    @Override
    public boolean isTouchingWaterOrRain() {
        return player.isTouchingWaterOrRain();
    }

    @Override
    public boolean isSneaking() {
        return player.isSneaking();
    }

    @Override
    public boolean isSwimming() {
        return player.isSwimming();
    }

    @Override
    public boolean isInvisible() {
        return player.isInvisible();
    }

    @Override
    public double[] getPosition() {
        return new double[]{player.getPos().getX(), player.getPos().getY(), player.getPos().getX()};
    }

    @Override
    public double getPosX() {
        return player.getPos().getX();
    }

    @Override
    public double getPosY() {
        return player.getPos().getY();
    }

    @Override
    public double getPosZ() {
        return player.getPos().getZ();
    }

    @Override
    public float getPitch() {
        return player.getPitch();
    }

    @Override
    public float getYaw() {
        return player.getYaw();
    }

    @Override
    public float getSpeed() {
        return player.getSpeed();
    }

    @Override
    public int getExperienceLevel() {
        return player.experienceLevel;
    }

    @Override
    public int getTotalExperience() {
        return player.totalExperience;
    }

    @Override
    public float getExperienceProgress() {
        return player.experienceProgress;
    }
}

File: src/main/kotlin/forpleuvoir/hiirosakura/client/feature/event/eventinterface/impl/ClientPlayerInterface.kt
package forpleuvoir.hiirosakura.client.feature.event.eventinterface.impl

import forpleuvoir.hiirosakura.client.feature.event.eventinterface.IClientPlayerInterface
import net.minecraft.client.network.ClientPlayerEntity
import net.minecraft.util.registry.Registry

/**
 * 客户端玩家接口实现
 *
 * @author forpleuvoir
 *
 * #project_name hiirosakura
 *
 * #package forpleuvoir.hiirosakura.client.feature.event.eventinterface.impl
 *
 * #class_name ClientPlayerInterface
 *
 * #create_time 2021/8/1 14:47
 */
class ClientPlayerInterface(private val player: ClientPlayerEntity) : IClientPlayerInterface {
	override val uuid: String
		get() = player.uuidAsString
	override val name: String
		get() = player.entityName
	override val health: Float
		get() = player.health
	override val maxHealth: Float
		get() = player.maxHealth
	override val healthProgress: Double
		get() = (health / maxHealth).toDouble()
	override val mainHandItemRegisterId: String
		get() = Registry.ITEM.getId(player.mainHandStack.item).toString()
	override val mainHandItemTranslationKey: String
		get() = player.mainHandStack.item.translationKey
	override val mainHandItemDisplayName: String
		get() = player.mainHandStack.name.string
	override val mainHandItemDurable: Int
		get() = mainHandItemMaxDurable - player.mainHandStack.damage
	override val mainHandItemMaxDurable: Int
		get() = player.mainHandStack.maxDamage
	override val mainHandItemDurableProgress: Double
		get() = mainHandItemDurable.toDouble() / player.mainHandStack.maxDamage
	override val isFireImmune: Boolean
		get() = player.isFireImmune
	override val isOnFire: Boolean
		get() = player.isOnFire
	override val isTouchingWaterOrRain: Boolean
		get() = player.isTouchingWaterOrRain
	override val isSneaking: Boolean
		get() = player.isSneaking
	override val isSwimming: Boolean
		get() = player.isSwimming
	override val isInvisible: Boolean
		get() = player.isInvisible
	override val position: DoubleArray
		get() = doubleArrayOf(player.pos.getX(), player.pos.getY(), player.pos.getX())
	override val posX: Double
		get() = player.pos.getX()
	override val posY: Double
		get() = player.pos.getY()
	override val posZ: Double
		get() = player.pos.getZ()
	override val pitch: Float
		get() = player.pitch
	override val yaw: Float
		get() = player.yaw
	override val speed: Float
		get() = player.speed
	override val experienceLevel: Int
		get() = player.experienceLevel
	override val totalExperience: Int
		get() = player.totalExperience
	override val experienceProgress: Float
		get() = player.experienceProgress
}